blobstore/blobstore.go

```diff
@@ -199,12 +201,40 @@ func ParseUpload(req *http.Request) (blobs map[string][]*BlobInfo, other url.Val
 			return nil, nil, err
 		}
 		bi.BlobKey = appengine.BlobKey(params["blob-key"])
+		charset := params["charset"]
+
 		if ctype != "message/external-body" || bi.BlobKey == "" {
 			if formKey != "" {
 				slurp, serr := ioutil.ReadAll(part)
 				if serr != nil {
 					return nil, nil, errorf("error reading %q MIME part", formKey)
 				}
+
+				// Handle base64 content transfer encoding. multipart.Part transparently
+				// handles quoted-printable, and no special handling is required for
+				// 7bit, 8bit, or binary.
+				ctype, params, err = mime.ParseMediaType(part.Header.Get("Content-Transfer-Encoding"))
+				if err == nil && ctype == "base64" {
+					slurp, serr = ioutil.ReadAll(base64.NewDecoder(
+						base64.StdEncoding, bytes.NewReader(slurp)))
```

Comment:
Can you add a test for a multi-line base64 payload? I think the standard is 76 bytes per line.

Reply:
Sure will do

Reply:
Done